### ISSUE TITLE
Latest Binary on Macos not running, Bad CPU type in executable

### ISSUE THREAD

**Issue by bentxt** [c3lang/c3c#1388]
### System
```
> sw_vers
ProductName:		macOS
ProductVersion:		14.6.1
BuildVersion:		23G93
> machine
x86_64h
```

### Installation Steps and Error
```
wget https://github.com/c3lang/c3c/releases/download/latest/c3-macos.zip
unzip c3-macos.zip
cd macos/
./c3c
exec: Failed to execute process './c3c': Bad CPU type in executable.
```

**Comment by bentxt** [comment]
PS.: Building from scratch (cmake build) works

**Comment by bentxt** [comment]
I think I found the error, the binary provided is only for ARM

```
cd c3/macos
file c3c
c3c: Mach-O 64-bit executable arm64
```

**Comment by lerno** [comment]
Yes, the CI unfortunately only builds the aarch64 binaries, and worse – it's because it relies on Homebrew which will only give you no-fat LLVM libraries. That means there is no trivial way to cross compile it. However, Homebrew will update with the latest C3 binaries usually and it's fairly straightforward to compile on one's own. I'd like to offer a wide range of precompiled binaries, but that means setting up quite a bit of infrastructure to compile everything. Unless someone picks up that task soon, it unfortunately will just have to wait.

**Comment by lerno** [comment]
Do we want to fix this? @ManuLinares do we have x64 LLVM binaries for Mac now?

**Comment by Darvisim** [comment]
> Do we want to fix this? [@ManuLinares](https://github.com/ManuLinares) do we have x64 LLVM binaries for Mac now?

iirc we do have x64 LLVM binaries for mac, it uses this artifact: https://github.com/c3lang/llvm-for-c3/releases/latest/download/llvm-darwin-amd64-dbg.tar.xz

**Comment by ManuLinares** [comment]
> Do we want to fix this? [@ManuLinares](https://github.com/ManuLinares) do we have x64 LLVM binaries for Mac now?

cmake should pick up macos-amd64 automatically and build the correct c3c

So we can close this.
